### PR TITLE
Load form state and manage form throughout react component lifecycle.

### PR DESCRIPTION
**Pull request by TimMikeladze**

### CODE REVIEW

File: src/connectForm.js [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
+        loaded: false,
+      };
     }
-    componentDidMount() {
+    async componentDidMount() {
+      this.form = (isFunction(form) ? form : () => form)();
       this.unsubscribe = this.form.store.subscribe(() => {
         this.forceUpdate();
       });
+      if (isFunction(this.form.loadPromise)) {
```
Comment (alexyuly) [inline src/connectForm.js]: Why not `typeof this.form.loadPromise === 'function'`?

File: src/connectForm.js [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
-      });
+      if (this.state.loaded) {
+        const fields = this.form.getFields();
+        return React.cloneElement(Component, {
+          ...this.props,
+          fields,
+          form: this.form,
+        });
+      }
+      return null;
```
Comment (alexyuly) [inline src/connectForm.js]: Minor purely stylistic nitpick - put your guard condition (`!this.state.loaded`) first. Not your main condition.

File: src/connectForm.js (hunk 1 above)
Comment (TimMikeladze) [inline src/connectForm.js]: Personal preference to use lodash util functions.

File: src/connectForm.js (hunk 2 above)
Comment (TimMikeladze) [inline src/connectForm.js]: Fixed.

**Pull request merged**